### PR TITLE
tests: net: lib: mqtt_publisher: move to new ztest API

### PR DESCRIPTION
**Pull request by hunterhu3000**
Move net mqtt_publisher tests to use new ztest API

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

### CODE REVIEW

**Review by rlubos — DISMISSED** [review]

**Review by enjiamai — CHANGES_REQUESTED** [review]
Hi @hunterhu3000 , you need to add CONFIG_ZTEST_NEW_API=y into the prj_tls.conf also to fix the build failure.

**Comment by hunterhu3000** [comment]
> Hi @hunterhu3000 , you need to add CONFIG_ZTEST_NEW_API=y into the prj_tls.conf also to fix the build failure.

@enjiamai Thank you for helping me fix the build issue. It seems the CI can pass now.

**Review by enjiamai — DISMISSED** [review]

**Comment by hunterhu3000** [comment]
It shows 
```
devicetree error: 'label' is marked as deprecated in 'properties:' in /home/ztest/zephyrproject/zephyr/dts/bindings/sdhc/zephyr,sdhc-spi-slot.yaml for node /soc/spi@4002d000/sdhc@0.
```
Remove the label in boards/arm/frdm_k64f/frdm_k64f.dts can fix the issue. But the background for it is unknown. Thanks to review.

File: boards/arm/frdm_k64f/frdm_k64f.dts [hunk 1]
```diff
@@ -156,7 +156,6 @@ arduino_spi: &spi0 {
 		compatible = "zephyr,sdhc-spi-slot";
 		reg = <0>;
 		status = "okay";
-		label = "SDHC_0";
 		mmc {
```
Comment (enjiamai) [inline boards/arm/frdm_k64f/frdm_k64f.dts:159]: Hi @hunterhu3000 , suggest to put this fix in another commit, or even another PR.

Comment (hunterhu3000) [reply boards/arm/frdm_k64f/frdm_k64f.dts]: got it. Thanks
A new PR without this change is submitted. Thanks to review.

**Review by enjiamai — DISMISSED** [review]

**Review by rlubos — DISMISSED** [review]

**Review by enjiamai — APPROVED** [review]

**Review by rlubos — APPROVED** [review]
